### DISCUSSION TITLE
Parsing data in useInfiniteQuery while also keeping the paginated results?

### DISCUSSION THREAD

**Discussion by ShadiestGoat** [TanStack/query#4867] [Q&A] [answered]
Hello, I'm quite new to tanstack query, going through svelte query, so I'm not sure if this is possible or not. Basically, I have a setup where my backend returns paginated data, `post[]`. I also have a websocket, which would sometimes give updated information specific to a post, based on post id. 

I still need the paginated results for displaying and such, but I also want to change specific posts based on id. So, I'm asking - is this at all possible? Is there a something simple I'm missing?

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

Have you tried programmatically setting the query data via the query client? Please see: https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientsetquerydata

**Reply by ShadiestGoat** [reply]
Havent seen that one, thanks!

**Reply by louis-young** [reply]
No worries! Let us know if that helps you achieve what you're looking for 🙂
